RELEASING.md

```diff
@@ -76,16 +76,6 @@ Then run `chmod 600 ./.pypirc` so only you can read/write.
 
 1. Update version to, e.g. 0.9.1+dev in `snorkel/version.py`.
 
-1. Add a new changelog entry for the unreleased version:
```

Comment:
We do actually still want to create these sections under `[Unreleased]` so that there's a place to being storing these different types of changes as they're merged into the tip of main. We just don't want to have both `[Unreleased]` _and_ the name of the next version.

Reply:
Ok added back